### DISCUSSION TITLE
[react-aria] useMove: How to access initial event coordinates?

### DISCUSSION THREAD

**Discussion by weevolve-dev** [adobe/react-spectrum#2584] [Q&A] [answered]
I want to implement a selection rectangle component that uses the useMove hook to track the pointer position while the user is drawing the rectangle.

The callback parameters of onMove, onMoveStart and onMoveEnd only provide me with x/y coordinates relative to the position where the move gesture started, but where can I find the screen coordinates of this starting point? Is there a way to access the initial click/touch event, for example?

**Comment by reidbarber** [comment, marked as answer]
This is outside of the scope of `useMove`, but there are a probably a few ways you could do this. If you want the movable element's position relative to the viewport, you could use [Element.getBoundingClientRect()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) like this:

```jsx
let objectRef = useRef(null);

/*
    x: ...
    y: ...
    width: ...
    height: ...
    top: ...
    right: ...
    bottom: ...
    left: ...
*/
console.log(objectRef.current.getBoundingClientRect());

<div
    ref={objectRef}
    {...moveProps}
...
```

I modified the useMove docs' example to log start position (relative to viewport) of each move here:
https://codesandbox.io/s/usemove-viewport-position-1vu02?file=/src/App.js

**Reply by snowystinger** [reply]
Might be a similar use case to https://github.com/adobe/react-spectrum/issues/2031, we could consider it or at least consider adding to documentation something similar to what @reidbarber suggested

**Reply by weevolve-dev** [reply]
thanks @reidbarber for the clarification. Unfortunately, your proposed solution does not work for me because in my case, the user does not press an element at the beginning of the move interaction - I have to create an element (the selection rectangle) at the initial pointer position. 

But anyway, now that I know that it is out of scope, I will find another workaround, thanks!

**Reply by snowystinger** [reply]
If you need something right this minute, you could add your own onPointerDown/onKeyDown and use mergeProps(yourHandlers, moveProps) and spread that on the target element. Then you can just access the event properties you need without interfering with useMove.
Couple things to think about, how are you going to expose this to a keyboard user? a keyboard user would need a tab stop to get started from.
How are you going to expose this to screen readers. VoiceOver and other screen readers, report the events in the very center of the target element typically.

**Reply by weevolve-dev** [reply]
@snowystinger good points, so I maye need to re-think the design of my feature. And thanks for pointing me to mergeProps, I was not aware about it
